compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -1845,6 +1851,18 @@ public BLangLetVariable createLetVariable(LetVariableDeclarationNode letVarDecl)
         return letVar;
     }
 
+    @Override
+    public BLangNode transform(CaptureBindingPatternNode captureBindingPatternNode) {
+        Node parent = captureBindingPatternNode.parent();
+        if (parent instanceof FieldBindingPatternVarnameNode || parent instanceof FieldBindingPatternFullNode
```

Comment:
can we use the kind please.

Reply:
Kind is the same for both of these :( So went for instanceof. Shall I change the kind as well?

Reply:
@SupunS shouldn't we have different kinds?

Reply:
Fixed

Reply:
Seems like only `FieldBindingPatternVarnameNode` and `FieldBindingPatternFullNode` has `FIELD_BINDING_PATTERN` kind. So we can go with kind is guess.